### ISSUE TITLE
Reword GD4 and HD17

### ISSUE THREAD

**Issue by bzah** [clix-meta/clix-meta#64]
# Please explain briefly what is wrong

Definitions of Growing degree days (GD4) and Heating degree days (HD17) seem to be incorrect.

In clix-meta, both indices are described as a sum of raw values that are either above or below a threshold ( > 4ºC and < 17ºC):
> long_name: "Heating degree days (sum of Tmean < 17 C)"
> long_name: "Growing degree days (sum of Tmean > 4 C)"

However, according to [ECAD ATBD](https://knmi-ecad-assets-prd.s3.amazonaws.com/documents/atbd.pdf) the definitions are:
> Heating degree days (sum of 17 ◦C - TG) (◦C)
<img width="247" alt="image" src="https://user-images.githubusercontent.com/16700639/182612332-62cd6834-a5b4-48a0-b271-8dc9b0bc7089.png">

And

> Growing degree days (sum of TG > 4 ◦C) (◦C)
<img width="340" alt="image" src="https://user-images.githubusercontent.com/16700639/182612221-df7d6b49-7818-43e4-a471-90956d739604.png">

So I think there are several issues:
1. in hd17, there is no threshold filtering, the `17ºC - x` is supposed to be applied to every values (not true on HDDheatn!).
2. in gd4, the long_name does not properly express the `x - 4ºC` operation, it would seem it's just a sum of values above 4ºC.
3. The properties in `index_function` _might_ be wrong (not sure about that, because I don't use these).

# Suggest specific changes to the `index_definition` table

> long_name: "Heating degree days (sum of 17 C - Tmean)"
> long_name: "Growing degree days (sum of Tmean - 4 C when Tmean > 4 C )"

**Comment by larsbarring** [comment]
@bzah thanks for these comments and suggestions!

Regarding your specific comments
1. This would mean that for a location where daily mean temperature (TG) is varying around the mean of 17 C will have `hd17`=0. Can you confirm that this is indeed what is intended with `hd17` (and just not a mistake in ATBD). In that case also the `standard name` have to be changed, and I do not think a suitable one exists.
2. I am not sure I understand. Could you please explain.
3. If you are right regarding the first point we definitely have to change also the index function specification for `hd17`. It might be the same also for `gd4`.

**Comment by bzah** [comment]
1. I agree that it looks like an oversight in ATBD. Maybe this should be reported to the ECAD team ?
I can't confirm anything on the usage of HD17 as I don't do any analysis myself, but I can say that both xclim and climpact implementations are **not** following ECAD definition.
_Note: I just remember that [I already had this discussion on xclim](https://github.com/Ouranosinc/xclim/issues/952)... So it seems it's indeed a mistake in ATBD._

Also, about the standard name of HD17, I believe it's not an excess but a deficit so `integral_wrt_time_of_air_temperature_deficit` might be more suited.

2. My issue with GD4 is that the long_name property `Growing degree days (sum of Tmean > 4 C)` make it look like the operations to compute this index are:
```
x = data > 4ºC
gd4 = sum(x)
```
Where the formula and the standard_name properly express the _excess_ calculation:
```
x = data > 4ºC
y = 4ºC - x
gd4 = sum(y)
```

IMHO, long_name should be at least as explicit as the standard_name and `x-4` should written in it somewhere.
It's kind of a big deal to me because I'm using these long_name(s) to generate the documentation of our generated API in icclim.

**Comment by bzah** [comment]
Furthermore, the `reference` for HD17 could be `ECA&D modified` (or something like that) until the ATBD is fixed.

**Comment by larsbarring** [comment]
As per offline email from the ECA&D team there was a mistake in the ATBD description. This is now corrected.

**Comment by larsbarring** [comment]
Thanks for raising this issue @bzah! 

It helped to improve the index documentation over at ECA&D, as well as sorting out the standard name mistake in the master table. 

Regarding the *long names* I agree that they should be as explicit as possible, but also short and non-technical aiming at a wide audience (think of a generic graph title or a table heading.) Thus I changed to ones you point at, but for a couple of closely related indices I used a different wording. Maybe others might have suggestions towards this end.

There is now a PR in the works, but it will take couple of weeks until it is merged.
